esmvaltool/namelists/namelist_preprocessor_derive_test.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+models:
+#  - {model: CESM1-WACCM,    project: CMIP5, mip: Amon, exp: historical, ensemble: r2i1p1, start_year: 1997, end_year: 2005}
+#  - {model: CNRM-CM5,       project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: GFDL-CM3,       project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: GISS-E2-H,      project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p2, start_year: 1997, end_year: 2005}
+  - {model: GISS-E2-R,      project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p2, start_year: 1997, end_year: 2005}
+#   - {model: MIROC-ESM-CHEM, project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: ESACCI-OZONE,   project: OBS,   tier: 2,   type: sat,       version: L3,      start_year: 1997, end_year: 2005}
+#   - {model: NIWA,           project: OBS,   tier: 3,   type: reanaly,   version: 1,       start_year: 1997, end_year: 2005}
+
+preprocessors:
+
+  preprocessor:
+    derive: true
+
+diagnostics:
+
+  derive_diagnostic:
+    description: Test variable derivation
+    variables:
+      toz:
+        preprocessor: preprocessor
+        field: T2Ms
```

Comment:
It will be nice to have a way of forcing to use the derived version even if the data is present. Sometimes you don know if the variable has been computed in a sensible way

Reply:
If the variable is directly available (usually in the observations), we should use it and not derive it. Also because the primary variables to derive it will not be available.

Reply:
Yes, I know. But I am thinking about models here and there are some cases in which forcing derivation  may be of interest for the user. Let's discuss and example for the wind velocity module (sfcWind).

Suppose that you are interested in a diagnostic for daily means of sfcWind. Sometimes you will have the variable, sometimes you will have to derive it from uas and vas (the components). The bad news is that the daily average of the module is not the module of the daily averages of the components. But you have doubts that this effect is strong enough to affect your results in a significant way. The easiest way to address this is to run the diagnostic two times: one with the stored sfcWind and one with the derivation and compare the results. I know that this can be done with ESMValTool just working with the data paths so it can't find sfcWind but, if possible, it would be nice to add a flag to simplify user's life

Reply:
Thanks for the example. I think the default behaviour should be "do not derive if data available", but we could add such a flag (e.g., `force_derivation = true`) for some special cases, like the one you mentioned. This flag should probably be part of the `variable` dictionary, right?

Reply:
Yes, that probably makes most sense. Do you want me to add it in this pull request or make a separate issue?

Reply:
You can add it in this one, I will have probably no time for testing it till Monday anyway. Thanks!

Reply:
done

Reply:
Thanks!